src/graph_compute/chromatic_scheduling.h

```diff
@@ -0,0 +1,105 @@
+#ifndef CHROMATIC_SCHEDULING_H_
+#define CHROMATIC_SCHEDULING_H_
+
+#if D1_CHROM
+
+#include <algorithm>
+#include "./common.h"
+
+struct scheddata_t {
+  vid_t cntColors;
+  vid_t * cntNodesPerColor;
+  vid_t * nodesByColor;
+};
+typedef struct scheddata_t scheddata_t;
+
+struct sched_t {
+  vid_t color;
+};
+typedef struct sched_t sched_t;
+
+#include "./update_function.h"
+
+static inline vid_t colorGraph(vertex_t * const nodes,
+                              const vid_t cntNodes) {
+  bool * colors = new (std::nothrow) bool[cntNodes+1];
+  vid_t maxColor = 0;
+  for (vid_t v = 0; v < cntNodes; v++) {
+    for (vid_t edge = 0; edge <= nodes[v].cntEdges; edge++) {
+      colors[edge] = false;
+    }
+    for (vid_t edge = 0; edge < nodes[v].cntEdges; edge++) {
+      if (nodes[v].edges[edge] < v) {
+        colors[edge] = true;
+      }
+    }
+    static const vid_t MAX_COLOR = cntNodes + 1;
```

Comment:
Maybe rename this to `NO_COLOR`? `MAX_COLOR` is misleading because you can't have more colors than nodes.